aoc/internal/array.go

```diff
@@ -0,0 +1,45 @@
+package internal
```

Comment:
I make this mistake all the time by calling them arrays. But your operations are on slice, so perhaps name the file slice.go to avoid upsetting hardcore go devs (not me).